pkg/apis/networking/v1alpha1/service_types.go

```diff
@@ -0,0 +1,99 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:shortName=mcs,categories={karmada-io}
+
+// MultiClusterService is a named abstraction of multi-cluster software service.
+// The name field of MultiClusterService is the same as that of Service name.
+// Services with the same name in different clusters are regarded as the same
+// service and are associated with the same MultiClusterService.
+// MultiClusterService can control the exposure of services to outside multiple
+// clusters, and also enable service discovery between clusters.
+type MultiClusterService struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the desired state of the MultiClusterService.
+	// +optional
+	Spec ServiceSpec `json:"spec,omitempty"`
+
+	// Status is the current state of the MultiClusterService.
+	// +optional
+	Status corev1.ServiceStatus `json:"status,omitempty"`
+}
+
+// ServiceSpec is the desired state of the MultiClusterService.
+type ServiceSpec struct {
```

Comment:
Please use the full name `MultiClusterServiceSpec` to avoid confusion between Kubernetes `ServiceSpec`.

Reply:
Updated.